src/ProjectFileIO.cpp

```diff
@@ -1757,38 +1835,121 @@ bool ProjectFileIO::WriteDoc(const char *table,
       return false;
    }
 
-   const MemoryStream &dict = autosave.GetDict();
+   const MemoryStream& dict = autosave.GetDict();
    const MemoryStream& data = autosave.GetData();
 
    // Bind statement parameters
    // Might return SQL_MISUSE which means it's our mistake that we violated
    // preconditions; should return SQL_OK which is 0
-   if (sqlite3_bind_blob(stmt, 1, dict.GetData(), dict.GetSize(), SQLITE_STATIC) ||
-       sqlite3_bind_blob(stmt, 2, data.GetData(), data.GetSize(), SQLITE_STATIC))
+   if (
+      sqlite3_bind_zeroblob(stmt, 1, dict.GetSize()) ||
+      sqlite3_bind_zeroblob(stmt, 2, data.GetSize()))
    {
       ADD_EXCEPTION_CONTEXT("sqlite3.query", sql);
       ADD_EXCEPTION_CONTEXT("sqlite3.rc", std::to_string(rc));
       ADD_EXCEPTION_CONTEXT("sqlite3.context", "ProjectGileIO::WriteDoc::bind");
 
-      SetDBError(
-         XO("Unable to bind to blob")
-      );
+      SetDBError(XO("Unable to bind to blob"));
       return false;
    }
 
+   const auto reportError = [this](auto sql) {
+      SetDBError(
+         XO("Failed to update the project file.\nThe following command failed:\n\n%s")
+            .Format(sql));
+   };
+
    rc = sqlite3_step(stmt);
+
    if (rc != SQLITE_DONE)
    {
       ADD_EXCEPTION_CONTEXT("sqlite3.query", sql);
       ADD_EXCEPTION_CONTEXT("sqlite3.rc", std::to_string(rc));
       ADD_EXCEPTION_CONTEXT("sqlite3.context", "ProjectGileIO::WriteDoc::step");
 
-      SetDBError(
-         XO("Failed to update the project file.\nThe following command failed:\n\n%s").Format(sql)
-      );
+      reportError(sql);
       return false;
    }
 
+   // Finalize the statement before commiting the transaction
+   sqlite3_finalize(stmt);
+   stmt = nullptr;
+
+   // Get rowid
+
+   int64_t rowID = 0;
+
+   const wxString rowIDSql =
+      wxString::Format("SELECT ROWID FROM %s.%s WHERE id = 1;", schema, table);
+
+   if (!Query(
+          rowIDSql,
+          [&rowID](int cols, char** vals, char** names)
+          {
+             if (cols != 1)
+                return -1;
+
+             rowID = std::stoll(vals[0]);
```

Comment:
Should rowID be long long?

Reply:
It's `int64_t` in SQLite. Usually, it's an alias to `long long` anyway. At least until we will have 128 bit `long long` :-)